File: models/privateChat.js
const mongoose = require('mongoose');

var privateChatSchema = mongoose.Schema({
    person1 = {type: mongoose.Schema.Types.ObjectId, ref: 'User'},
    person2 = {type: mongoose.Schema.Types.ObjectId, ref: 'User'},
    messages: [{
        message: {type: mongoose.Schema.Types.ObjectId, ref: 'Message'}
    }],
    createdAt: {type: Date, default: Date.now}
});

module.exports = mongoose.model('PrivateChat', privateChatSchema);

File: slabber_chat/lib/models/private_chat.dart
import 'user_model.dart';
import 'message_model.dart';

class PrivateChat{
  final User person1;
  final User person2;
  List<Message> messages;

  PrivateChat({
    this.person1,
    this.person2,
    this.messages,
  });
}